### PR TITLE
Enable SWIFT XRT API access

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR enables SWIFT XRT API access.

### CODE REVIEW

File: doc/mma.md [hunk 1]
```diff
@@ -19,7 +19,8 @@ The vast majority of follow-up instruments will require some form of authenticat
 * LCO: A user account must be made on https://lco.global/, at which point the authentication takes the form `{"PROPOSAL_ID": "your_proposal_ID", "API_TOKEN": "testtoken"}`.
 * LT: A proposal ID, username, and password are passed as `{"username": "username", "password": "password", "LT_proposalID": "your_proposal_ID"}`.
 * SLACK: As discussed further [here](./slack.html), slack information is pass as `{"slack_workspace": "XXX", "slack_channel": "YYY", "slack_token": "ZZZ"}`.
-* SWIFT: A username and password are passed as `{"username": "username", "secret": "password"}`.
+* SWIFT Triggering: A username and password are passed as `{"username": "username", "secret": "password"}`.
+* SWIFT XRT Reductions: A user account must be made with the Swift-XRT data products API platform (see https://www.swift.ac.uk/user_objects/register.php to register). The authentication then is the email of the user `"XRT_UserID": "swift_email"}` where swift_email is the email addressed used upon sign up.
```
Comment (guynir42) [inline doc/mma.md:23]:
```suggestion
* SWIFT XRT Reductions: A user account must be made with the Swift-XRT data products API platform (see https://www.swift.ac.uk/user_objects/register.php to register). The authentication then is the email of the user `"XRT_UserID": "swift_email"}` where swift_email is the email address used upon sign up.
```

File: skyportal/facility_apis/swift.py [hunk 2]
```diff
@@ -82,10 +88,142 @@ def _build_payload(self, request):
...
+        altdata = request.allocation.altdata
+
+        T0 = Time(request.payload["T0"], format='iso')
+        MET = Time('2001-01-01 00:00:00', format='iso')
+        Tdiff = (T0 - MET).jd * 86400
+
+        if "detornot" in request.payload and request.payload["detornot"]:
+            centroid = True
+        else:
+            centroid = False
```
Comment (guynir42) [inline skyportal/facility_apis/swift.py:132]:
```suggestion
        centroid = bool(request.payload.get("detornot", False))
```

File: skyportal/facility_apis/swift.py [hunk 3]
```diff
@@ -82,10 +88,142 @@ def _build_payload(self, request):
...
+
+
 class UVOTXRTAPI(FollowUpAPI):
 
     """An interface to Swift operations."""
 
+    @staticmethod
+    def get(request, session):
+
+        """Get an analysis request result from Swift.
```
Comment (guynir42) [inline skyportal/facility_apis/swift.py:163]: analysis or followup request?

**Review by guynir42 — APPROVED** [review]
I have minor comments.

File: skyportal/facility_apis/swift.py (hunk 3 above)
Comment (mcoughlin) [reply skyportal/facility_apis/swift.py:160]: It is a reduction, so I think it is the right phrasing.

**Pull request merged**